ci/unit_tests/functions_deploy/main_test.py

```diff
@@ -0,0 +1,198 @@
+        functions_deploy.main(*args, **kwargs)
+
+    def _getFunctionsInPackage(self, package):
+        functionListUrl = self.cloudFunctionsUrl + '/' + self.urlNamespace + '/actions/?limit=0&skip=0'
+
+        functionListResp = requests.get(functionListUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                               os.environ['CLOUD_FUNCTIONS_PASSWORD']),
+                                        headers={'accept': 'application/json'})
+
+        assert functionListResp.status_code == 200
+
+        functionListJson = functionListResp.json()
+        functionNames = []
+
+        for function in functionListJson:
+            if (self.namespace + '/' + package) in function['namespace']:
+                functionNames += [function['name']]
+
+        return functionNames
+
+
+    def setup_method(self):
+        self.package = self.packageBase + str(uuid.uuid4())
+        self.packageCreated = False # test should set that to true if it created package for cloud functions
+
+    def teardown_method(self):
+        if self.packageCreated:
+            # get all functions in package and remove them
+            functionNames = self._getFunctionsInPackage(self.package)
+            for functionName in functionNames:
+                functionDelUrl =  self.cloudFunctionsUrl + '/' + self.urlNamespace + '/actions/' + self.package + \
+                    '/' + functionName
+
+                functionDelResp = requests.delete(functionDelUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                                        os.environ['CLOUD_FUNCTIONS_PASSWORD']))
+                assert functionDelResp.status_code == 200
+
+            # remove cloud function package
+            packageDelUrl = self.cloudFunctionsUrl + '/' + self.urlNamespace + '/packages/' + self.package
+
+            packageDelResp = requests.delete(packageDelUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                                  os.environ['CLOUD_FUNCTIONS_PASSWORD']))
+            assert packageDelResp.status_code == 200
+
+
+    def test_functionsUploadFromDirectory(self):
+        """Tests if functions_deploy uploads all supported functions from given directory."""
+
+        params = ['-c', os.path.join(self.dataBasePath, 'exampleFunctions.cfg'),
+                  '--cloudfunctions_username', os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                  '--cloudfunctions_password', os.environ['CLOUD_FUNCTIONS_PASSWORD'],
+                  '--cloudfunctions_package', self.package, '--cloudfunctions_namespace', self.urlNamespace,
+                  '--cloudfunctions_url', self.cloudFunctionsUrl]
+
+        # upload functions
+        self.t_noException([params])
+        self.packageCreated = True
+
+        # obtain list of uploaded functions
+        functionNames = self._getFunctionsInPackage(self.package)
```

Comment:
We could probably save one call to CF by setting functionNames as class object member and test in teardown_method it this is already set, then do not call _getFunctionsInPackage again.

Reply:
I want to have fully separate code in teardown method, just in case that something wrong happens in test